src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/RefDictionary.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Experimental.Collections
+{
+    public class RefDictionary<TKey, TValue> : IReadOnlyCollection<KeyValuePair<TKey, TValue>> where TKey : IEquatable<TKey> // almost IReadOnlyDictionary<TKey, TValue>
+    {
+        private int[] _buckets;
+        private Entry[] _entries;
+
+        private struct Entry
+        {
+            public TKey key;
+            public TValue value;
+            public int next;
+        }
+
+        public RefDictionary(int capacity = 16)
+        {
+            _buckets = new int[capacity];
```

Comment:
The capacity should be aligned up to be prime number to get a good distribution of hashcodes (same as in regular Dictionary).

Reply:
I've done this but had to copy the HashHelpers class and put it at the bottom of the file. How should this be accessed or duplicated?

Reply:
Suggest you put in same location being used by https://github.com/dotnet/corefxlab/pull/2525